### PR TITLE
add custom ID to add cron job

### PR DESCRIPTION
**Pull request by Licoy**
add custom ID to add cron job

### CODE REVIEW

File: cron.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ type Schedule interface {
 }
 
 // EntryID identifies an entry within a Cron instance
-type EntryID int
+type EntryID int64
```
Comment (pradykaushik) [inline cron.go:47]: why the change to `int64`?
`int` is more generic and platform dependent.

File: cron.go [hunk 2]
```diff
@@ -142,6 +142,22 @@ func (c *Cron) AddFunc(spec string, cmd func()) (EntryID, error) {
 	return c.AddJob(spec, FuncJob(cmd))
 }
 
+// AddFuncCustomId add a cron with a custom ID
+func (c *Cron) AddFuncCustomId(id EntryID, spec string, cmd func()) error {
```
Comment (pradykaushik) [inline cron.go:146]: what if the provided 'id' is not unique?
you need to vet the 'id' and make sure that it is unique.

File: cron.go [hunk 3]
```diff
@@ -142,6 +142,22 @@ func (c *Cron) AddFunc(spec string, cmd func()) (EntryID, error) {
...
+	job := FuncJob(cmd)
+	return c.AddJobCustomId(id, spec, job)
+}
+
+// AddJobCustomId add a cron with a custom ID
+func (c *Cron) AddJobCustomId(id EntryID, spec string, job Job) error {
```
Comment (pradykaushik) [inline cron.go:152]: what if the provided 'id' is not unique?
you need to vet the 'id' and make sure that it is unique.

File: cron.go (hunk 2 above)
Comment (pradykaushik) [reply cron.go:146]: This translates to a lack of test coverage.
Please include tests to check expected functionality when 'id's are not unique (two jobs having the same 'id').

**Review by pradykaushik — COMMENTED** [review]
I don't have much authority here but I saw this pull request and thought about mentioning a few things.

File: cron.go (hunk 1 above)
Comment (Licoy) [reply cron.go:47]: `int` range is a little small, for`int64` you can use snowflake id

File: cron.go (hunk 2 above)
Comment (Licoy) [reply cron.go:146]: The same as above. If this method is used, the external ID manager should be used to ensure the global uniqueness

File: cron.go (hunk 3 above)
Comment (Licoy) [reply cron.go:152]: The same as above. If this method is used, the external ID manager should be used to ensure the global uniqueness

**Comment by Licoy** [comment]
It seems that this PR is not applicable to the public, so close it first.
